apps/frontend/src/app/(app)/space/reading-list/page.tsx

```diff
@@ -1,5 +1,218 @@
+
 const ReadingListPage: React.FC = () => {
-  return <div className="text-gray-color p-16">reading list page</div>
+  const { session } = useAuth()
+  const { spaces } = useSpace() || { spaces: [] }
+  const {
+    readingItems,
+    fetchReadingList,
+    addItem: addItemToStore,
+    deleteItem: deleteItemFromStore,
+  } = useReadingStore()
+
+  const [loading, setLoading] = useState(true)
+  const [blockId, setBlockId] = useState<string | null>(null)
+
+  useEffect(() => {
+    const loadReadingList = async () => {
+      setLoading(true)
+      const readingListSpace = spaces.find(
+        (space) => space.name.toLowerCase() === "reading list"
+      )
+      if (readingListSpace) {
+        const fetchedBlockId = readingListSpace.blocks[0]
+        setBlockId(fetchedBlockId)
+        await fetchReadingList(session, fetchedBlockId)
+      }
+      setLoading(false)
+    }
+
+    if (spaces.length > 0 && !blockId) {
+      loadReadingList()
+    }
+  }, [spaces, session, fetchReadingList, blockId])
+
+  const [addingItem, setAddingItem] = useState(false)
+  const [newItemTitle, setNewItemTitle] = useState("")
+  const [newItemUrl, setNewItemUrl] = useState("")
+  const [newItemDescription, setNewItemDescription] = useState("")
+  const textareaRefTitle = useRef<HTMLTextAreaElement>(null)
+  const textareaRefUrl = useRef<HTMLTextAreaElement>(null)
+  const textareaRefDescription = useRef<HTMLTextAreaElement>(null)
+
+  useEffect(() => {
+    const textarea = textareaRefTitle.current
+    if (textarea) {
+      textarea.style.height = "auto"
+      textarea.style.height = `${textarea.scrollHeight}px`
+    }
+  }, [newItemTitle])
+
+  useEffect(() => {
+    const textarea = textareaRefUrl.current
+    if (textarea) {
+      textarea.style.height = "auto"
+      textarea.style.height = `${textarea.scrollHeight}px`
+    }
+  }, [newItemUrl])
+
+  useEffect(() => {
+    const textarea = textareaRefDescription.current
+    if (textarea) {
+      textarea.style.height = "auto"
+      textarea.style.height = `${textarea.scrollHeight}px`
+    }
+  }, [newItemDescription])
+
+  const addItem = async () => {
+    if (!newItemTitle || !newItemUrl || !blockId) return
+
+    try {
+      await addItemToStore(
+        session,
+        blockId,
+        newItemTitle,
+        newItemUrl,
+        newItemDescription
+      )
+      setNewItemTitle("")
+      setNewItemUrl("")
+      setNewItemDescription("")
+      setAddingItem(false)
+    } catch (error) {
+      console.error("Error adding item:", error)
+    }
+  }
+
+  const deleteItem = async (itemId: string) => {
+    if (!blockId) return
+
+    try {
+      await deleteItemFromStore(session, blockId, itemId)
+    } catch (error) {
+      console.error("Error deleting item:", error)
+    }
+  }
+
+  if (loading) {
+    return (
+      <div className="text-secondary-foreground p-16">
+        Loading reading list...
+      </div>
+    )
+  }
+
```

Comment:
The component is doing too much. It handles fetching, adding, and deleting items, as well as UI rendering.
It will be better t divide things to separate components!